src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -337,17 +337,15 @@ func UsernameMatchesMultipleExclusiveNamespaces(
 ) bool {
 	// Check namespaces and see if more than one match
 	matchCount := 0
+	userID := userutil.MakeUserID(username, cfg.Matrix.ServerName)
 	for _, appservice := range cfg.Derived.ApplicationServices {
-		for _, namespaceSlice := range appservice.NamespaceMap {
-			for _, namespace := range namespaceSlice {
-				// Check if we have a match on this username
-				if namespace.RegexpObject.MatchString(username) {
-					matchCount++
-				}
+		if appservice.IsInterestedInUserID(userID) {
+			if matchCount++; matchCount > 1 {
```

Comment:
Ermh, won't this always be True?

Reply:
It will be true on the second iteration.
The first match sets `matchCount` to 1 and since the second match it is `> 1`